nat-lab/tests/utils/netcat.py

```diff
@@ -0,0 +1,220 @@
+import asyncio
+from config import LIBTELIO_BINARY_PATH_MAC_VM, LIBTELIO_BINARY_PATH_WINDOWS_VM
+from contextlib import asynccontextmanager
+from datetime import datetime
+from typing import Optional, AsyncIterator
+from utils.connection import Connection, TargetOS
+from utils.output_notifier import OutputNotifier
+from utils.process import Process
+from utils.python import get_python_binary
+
+
+def _get_netcat_script_path(connection: Connection) -> str:
+    if connection.target_os == TargetOS.Windows:
+        return LIBTELIO_BINARY_PATH_WINDOWS_VM + "netcat.py"
+    if connection.target_os == TargetOS.Mac:
+        return LIBTELIO_BINARY_PATH_MAC_VM + "netcat.py"
+    return "nc"
+
+
+class NetCat:
+    """Wrapper class for the NC command"""
+
+    def __init__(
+        self,
+        connection: Connection,
+        host_ip: Optional[str],
+        port: int,
+        listen: bool = False,  # l-flag
+        udp: bool = False,  # u-flag
+        ipv6: bool = False,  # 6-flag
+        detached: bool = False,  # d-flag
+        port_scan: bool = False,  # z-flag
+        source_ip: Optional[str] = None,  # s-flag
+    ):
+        """
+        Create an instance of the nc command
+
+        Parameters
+        ----------
+            connection : Connection
+                Target connection to run the command on
+            port : int
+                Port number to connect to
+            host_ip : str | None
+                IP address of the host to connect to, not necessery when the listen option is used
+            listen : bool
+                Listen for an incoming connection rather than initiate a connection to a remote host
+            udp : bool
+                Use UDP instead of TCP protocol
+            ipv6 : bool
+                Force to use ipv6 addresses only
+            detached : bool
+                Do not attempt to read from stdin
+            port_scan : bool
+                Scan for listening daemons, without sending any data
+            source_ip : str | None
+                Specifies the IP of the interface which is used to send the packets
+        """
+        flags = "-nv"  # don't do any dns lookups and enable vebose output
+        flags += "6" if ipv6 else "4"
+        if listen:
+            flags += "l"
+        if udp:
+            flags += "u"
+        if detached:
+            flags += "d"
+        if port_scan and not listen:
+            flags += "z"
+
+        # use the built in netcat command on linux
+        if connection.target_os == TargetOS.Linux:
+            command = ["nc", flags, str(port)]
+        else:
```

Comment:
I think this if/else magic should be in _get_netcat_script_path. Now you have it in two places

Reply:
Good idea!